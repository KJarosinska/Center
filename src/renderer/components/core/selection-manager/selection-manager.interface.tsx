import { ChangeEvent, ReactElement } from "react"
import { Size } from "Renderer/components/core/input-checkbox/input-checkbox.component"

export interface SelectionManagerProps {
  checkboxSize?: Size
  selectedItemsNumber: number
  allItemsSelected: boolean
  messageId: string
  buttons?: ReactElement[]
  enlarged?: boolean
  onToggle?: (e?: ChangeEvent) => void
}
