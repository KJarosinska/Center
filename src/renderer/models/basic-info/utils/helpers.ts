import { SimCard } from "Renderer/models/basic-info/interfaces"

export const getActiveNetworkFromSim = (simCards: SimCard[] = []) => {
  const activeSimCard = simCards.filter(simCard => simCard.active)
  if (activeSimCard.length === 0) {
    return "No connection"
  }
  return activeSimCard[0].network
}
