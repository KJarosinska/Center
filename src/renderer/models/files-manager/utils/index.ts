import { FilesManagerState } from "Renderer/models/files-manager/files-manager.interface"

export const prepareDataForStackedBarChart = (data: FilesManagerState) => {
  return data.memoryData.map(el => {
    const { filesType, filesAmount, occupiedMemory, ...rest } = el
    return {
      value: occupiedMemory,
      ...rest,
    }
  })
}
