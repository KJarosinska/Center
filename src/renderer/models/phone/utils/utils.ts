import Faker from "faker"

// TODO: remove before production
export const generateFakeData = (numberOfContacts: number) => {
  return Array(numberOfContacts)
    .fill(0)
    .map(_ => ({
      id: Faker.random.uuid(),
      firstName: Faker.name.firstName(),
      lastName: Faker.name.lastName(),
      phoneNumber: Faker.phone.phoneNumber(),
      favourite: Faker.random.boolean(),
    }))
}

export const generateSortedStructure = (fakeState: any) => {
  const alphabet = new Array(26)
    .fill(1)
    .map((_, i) => String.fromCharCode(65 + i))

  const sortedContactList = fakeState.sort((a: any, b: any) =>
    a.firstName.localeCompare(b.firstName)
  )

  const generateFakeStructure = () => {
    const fakeStructure = []

    for (const letter of alphabet) {
      fakeStructure.push({
        category: letter,
        contacts: [],
      })
    }

    fakeStructure.unshift({
      category: "Favs",
      contacts: [],
    })

    return fakeStructure
  }

  const placeContactsInStructure = () => {
    const fakeStructure: any = generateFakeStructure()
    fakeStructure.forEach((fakeContact: any) => {
      const contactLetter = fakeContact.category
      const contactList: string[] = fakeContact.contacts
      sortedContactList.forEach((sortedContact: any) => {
        if (
          sortedContact.firstName.charAt(0) === contactLetter &&
          !sortedContact.favourite
        ) {
          contactList.push(sortedContact)
        }
      })
    })

    sortedContactList.forEach((contacts: any) => {
      if (contacts.favourite) {
        fakeStructure[0].contacts.push(contacts)
      }
    })

    return fakeStructure
  }

  return placeContactsInStructure()
}

export const sanitizeContacts = (structure: any) => {
  return structure.filter((el: any) => el.contacts.length > 0)
}

export const filterContacts = (contacts: any, substring: string) => {
  return contacts.map((contactsByLetter: any) => ({
    ...contactsByLetter,
    contacts: contactsByLetter.contacts.filter((contact: any) => {
      const filterableFields = Object.values(contact).filter(value => {
        return typeof value === "string"
      })
      return filterableFields.some((value: any) =>
        value.toLowerCase().includes(substring.toLowerCase())
      )
    }),
  }))
}
