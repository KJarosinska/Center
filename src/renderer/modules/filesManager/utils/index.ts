import {
  FilesManagerData,
  FilesManagerState,
} from "Renderer/models/files-manager/files-manager.interface"

export const memoryDataWithConvertedBytes = (data: FilesManagerState) => {
  return data.memoryData.map((element: FilesManagerData) => ({
    ...element,
    occupiedMemory: element.occupiedMemory / Math.pow(1024, 3),
  }))
}
