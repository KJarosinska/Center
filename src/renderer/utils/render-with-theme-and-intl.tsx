import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import { LANGUAGE } from "Renderer/constants/languages"
import localeEn from "Renderer/locales/main/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={LANGUAGE.default}
        locale={LANGUAGE.default}
        messages={localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}

export function renderWithTheme(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithTheme<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithTheme<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  const originalConsoleError = console.error
  console.error = (...args: any) => {
    if (
      !(
        args[0].startsWith("[React Intl] Cannot format message:") ||
        args[0].startsWith("[React Intl] Missing message:")
      )
    ) {
      originalConsoleError.call(console, ...args)
    }
  }
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider locale={LANGUAGE.default}>{ui}</IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
