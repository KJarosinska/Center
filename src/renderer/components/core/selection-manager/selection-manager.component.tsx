import React from "react"
import { FormattedMessage } from "react-intl"
import InputCheckbox from "Renderer/components/core/input-checkbox/input-checkbox.component"
import { SelectionManagerProps } from "Renderer/components/core/selection-manager/selection-manager.interface"
import Text, {
  TextDisplayStyle,
} from "Renderer/components/core/text/text.component"
import {
  backgroundColor,
  borderColor,
  borderRadius,
} from "Renderer/styles/theming/theme-getters"
import FunctionComponent from "Renderer/types/function-component.interface"
import { noop } from "Renderer/utils/noop"
import styled from "styled-components"

const Buttons = styled.div`
  grid-area: Buttons;
  justify-self: end;
  display: grid;
  grid-auto-flow: column;
  grid-column-gap: 1.6rem;
  padding-right: 2.4rem;
`

const CheckboxWrapper = styled.div`
  grid-area: Checkbox;
`

const Checkbox = styled(InputCheckbox)``

const Info = styled(Text)`
  grid-area: Info;
  line-height: 1.1;
`

const SelectionManagerWrapper = styled.section<{ enlarged?: boolean }>`
  display: grid;
  grid-template-areas: "Checkbox Info Buttons";
  grid-template-columns: auto 1fr auto;
  grid-column-gap: 0.8rem;
  align-items: center;
  height: 4rem;
  padding: 0 0.7rem;
  padding-left: ${({ enlarged }) => (enlarged ? 1.4 : 0.7)}rem;
  background-color: ${backgroundColor("accent")};
  border-radius: ${borderRadius("medium")};
  border: solid 0.1rem ${borderColor("grey3")};
  box-sizing: border-box;

  ${Checkbox} {
    /* TODO: Remove below style and add a size prop for checkbox */
    transform: scale(${({ enlarged }) => (enlarged ? 1 : 0.7)});
  }
`

const SelectionManager: FunctionComponent<SelectionManagerProps> = ({
  className,
  selectedItemsNumber,
  allItemsSelected,
  messageId,
  buttons,
  enlarged,
  onToggle = noop,
}) => {
  return (
    <SelectionManagerWrapper className={className} enlarged={enlarged}>
      <CheckboxWrapper>
        <Checkbox
          checked
          indeterminate={!allItemsSelected}
          onChange={onToggle}
        />
      </CheckboxWrapper>
      <Info
        displayStyle={TextDisplayStyle.MediumFadedLightText}
        data-testid="info"
      >
        <FormattedMessage
          id={messageId}
          values={{ num: allItemsSelected ? -1 : selectedItemsNumber }}
        />
      </Info>
      {Boolean(buttons?.length) && (
        <Buttons data-testid="buttons">{buttons}</Buttons>
      )}
    </SelectionManagerWrapper>
  )
}

export default SelectionManager
