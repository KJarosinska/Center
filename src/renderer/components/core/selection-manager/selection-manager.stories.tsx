import { action } from "@storybook/addon-actions"
import { button, withKnobs } from "@storybook/addon-knobs"
import { storiesOf } from "@storybook/react"
import React, { useState } from "react"
import ButtonComponent, {
  DisplayStyle,
} from "Renderer/components/core/button/button.component"
import SelectionManager from "Renderer/components/core/selection-manager/selection-manager.component"
import { SelectionManagerProps } from "Renderer/components/core/selection-manager/selection-manager.interface"
import Text, {
  TextDisplayStyle,
} from "Renderer/components/core/text/text.component"
import Delete from "Renderer/svg/delete.svg"
import Export from "Renderer/svg/upload.svg"
import styled from "styled-components"

export const deleteButton = (
  <ButtonComponent
    key="delete"
    label={"Delete"}
    displayStyle={DisplayStyle.Link1}
    Icon={Delete}
    data-testid="button"
    onClick={action("delete")}
  />
)

export const exportButton = (
  <ButtonComponent
    key="export"
    label={"Export"}
    displayStyle={DisplayStyle.Link1}
    Icon={Export}
    data-testid="button"
    onClick={action("export")}
  />
)

const buttons = [exportButton, deleteButton]

const useSelectionManager = () => {
  const [itemsNumber, setItemsNumber] = useState()
  const allItems = itemsNumber === 10

  const onToggle = () => {
    alert("Toggled")
  }

  const selectAll = () => setItemsNumber(10)
  const selectOne = () => setItemsNumber(1)
  const selectFew = () => setItemsNumber(1 + Math.ceil(Math.random() * 8))

  button("select all items", selectAll)
  button("select 1 item", selectOne)
  button("select few items", selectFew)
  return { allItems, onToggle, itemsNumber }
}

export const PredefinedSelectionManager = ({
  selectedItemsNumber = 1,
  allItemsSelected = false,
  ...props
}: Partial<SelectionManagerProps>) => {
  return (
    <SelectionManager
      messageId="view.name.messages.conversations.selectionsNumber"
      selectedItemsNumber={selectedItemsNumber}
      allItemsSelected={allItemsSelected}
      {...props}
    />
  )
}

export const CustomizedPredefinedSelectionManager = styled(
  PredefinedSelectionManager
)`
  grid-template-columns: 4.2rem 1fr auto;
`

storiesOf("Components", module)
  .addDecorator(withKnobs)
  .add("SelectionManager", () => {
    const { allItems, itemsNumber, onToggle } = useSelectionManager()
    return (
      <div style={{ margin: "2rem" }}>
        <Text displayStyle={TextDisplayStyle.SmallText}>Basic</Text>
        <br />
        <PredefinedSelectionManager
          selectedItemsNumber={itemsNumber}
          allItemsSelected={allItems}
          onToggle={onToggle}
        />
        <br />
        <br />
        <Text displayStyle={TextDisplayStyle.SmallText}>
          Basic with custom styles
        </Text>
        <br />
        <CustomizedPredefinedSelectionManager
          selectedItemsNumber={itemsNumber}
          allItemsSelected={allItems}
          onToggle={onToggle}
        />
        <br />
        <br />
        <Text displayStyle={TextDisplayStyle.SmallText}>With buttons</Text>
        <br />
        <PredefinedSelectionManager
          selectedItemsNumber={itemsNumber}
          allItemsSelected={allItems}
          onToggle={onToggle}
          buttons={buttons}
        />
        <br />
        <br />
        <Text displayStyle={TextDisplayStyle.SmallText}>
          Enlarged with buttons
        </Text>
        <br />
        <PredefinedSelectionManager
          selectedItemsNumber={itemsNumber}
          allItemsSelected={allItems}
          onToggle={onToggle}
          buttons={buttons}
          enlarged
        />
      </div>
    )
  })
