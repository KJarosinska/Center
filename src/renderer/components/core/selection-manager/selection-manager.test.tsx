import "@testing-library/jest-dom/extend-expect"
import "jest-styled-components"
import React from "react"
import { SelectionManagerProps } from "Renderer/components/core/selection-manager/selection-manager.interface"
import {
  deleteButton,
  exportButton,
  PredefinedSelectionManager,
} from "Renderer/components/core/selection-manager/selection-manager.stories"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"

const renderSelectionManager = (props: Partial<SelectionManagerProps> = {}) => {
  const outcome = renderWithThemeAndIntl(
    <PredefinedSelectionManager {...props} />
  )
  return {
    ...outcome,
    getWrapper: () => outcome.container.querySelector("section"),
    getCheckbox: () => outcome.getByRole("checkbox"),
    getInfo: () => outcome.getByTestId("info"),
    getButtonsWrapper: () => outcome.queryByTestId("buttons"),
    getButtons: () => outcome.queryAllByTestId("button"),
  }
}

test("matches snapshot", () => {
  const { container } = renderSelectionManager()
  expect(container).toMatchSnapshot()
})

test("has checkbox element", () => {
  const { getCheckbox } = renderSelectionManager()
  expect(getCheckbox()).toBeInTheDocument()
})

test("has info element", () => {
  const { getInfo } = renderSelectionManager()
  expect(getInfo()).toBeInTheDocument()
})

test("has no buttons by default", () => {
  const { getButtonsWrapper, getButtons } = renderSelectionManager()
  expect(getButtonsWrapper()).not.toBeInTheDocument()
  expect(getButtons()).toHaveLength(0)
})

test("render buttons properly", () => {
  const { getButtonsWrapper, getButtons } = renderSelectionManager({
    buttons: [exportButton, deleteButton],
  })
  expect(getButtonsWrapper()).toBeInTheDocument()
  expect(getButtons()).toHaveLength(2)
  expect(getButtons()[0]).toHaveTextContent("Export")
  expect(getButtons()[1]).toHaveTextContent("Delete")
})

test("renders text info for one item properly", () => {
  const { getInfo } = renderSelectionManager({ selectedItemsNumber: 1 })
  expect(getInfo()).toHaveTextContent("1 Conversation selected")
})

test("renders text info for few items properly", () => {
  const { getInfo } = renderSelectionManager({ selectedItemsNumber: 4 })
  expect(getInfo()).toHaveTextContent("4 Conversations selected")
})

test("renders text info for all items properly", () => {
  const { getInfo } = renderSelectionManager({ allItemsSelected: true })
  expect(getInfo()).toHaveTextContent("All Conversations selected")
})

test("renders indeterminate status properly", () => {
  const { queryByTestId } = renderSelectionManager({ allItemsSelected: false })
  expect(queryByTestId("state-indeterminate")).toBeInTheDocument()
})

test("renders indeterminate status properly", () => {
  const { queryByTestId } = renderSelectionManager({ allItemsSelected: true })
  expect(queryByTestId("state-checked")).toBeInTheDocument()
})

test("renders expanded selection manager properly", () => {
  const { getWrapper, getCheckbox } = renderSelectionManager({ enlarged: true })
  expect(getWrapper()).toHaveStyleRule("padding-left", "1.4rem")
  expect(getCheckbox()).toHaveStyleRule("height", "2rem")
})
