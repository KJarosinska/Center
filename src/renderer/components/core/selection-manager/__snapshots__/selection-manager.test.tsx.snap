// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c6 {
  margin: 0;
  font-size: 1.4rem;
  line-height: 1.05;
  color: #6a6a6a;
  font-weight: 300;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 1.5rem;
  width: 0.8rem;
  position: relative;
}

.c4 svg {
  max-height: 100%;
  max-width: 100%;
  height: 100%;
  width: 100%;
}

.c2 {
  position: relative;
  height: 2rem;
  width: 2rem;
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  display: inline-block;
  height: 100%;
  width: 100%;
  background-clip: content-box;
  border: 0.1rem solid #cdcecf;
  border-radius: 0.2rem;
  background-color: #ffffff;
  background-color: #3b3f42;
  border-color: #3b3f42;
}

.c3:hover {
  border-color: #3b3f42;
  cursor: pointer;
  -webkit-transition: border-color 0.5s linear;
  transition: border-color 0.5s linear;
}

.c3:checked {
  background-color: #3b3f42;
  border-color: #3b3f42;
}

.c3:checked + svg {
  display: initial;
}

.c3 + svg {
  display: initial;
}

.c5 {
  display: none;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5 path {
  stroke: #ffffff;
}

.c1 {
  grid-area: Checkbox;
}

.c7 {
  grid-area: Info;
  line-height: 1.1;
}

.c0 {
  display: grid;
  grid-template-areas: "Checkbox Info Buttons";
  grid-template-columns: auto 1fr auto;
  grid-column-gap: 0.8rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 4rem;
  padding: 0 0.7rem;
  padding-left: 0.7rem;
  background-color: #f4f5f6;
  border-radius: 0.5rem;
  border: solid 0.1rem #e9e9e9;
  box-sizing: border-box;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="checkbox-wrapper"
        size="2"
      >
        <input
          checked=""
          class="c3"
          type="checkbox"
        />
        <div
          class="c4 c5"
          data-testid="icon-CheckIndeterminate"
          height="1.5"
          width="0.8"
        >
          <test-file-stub
            image="test-file-stub"
          />
        </div>
      </div>
    </div>
    <p
      class="c6 c7"
      data-testid="info"
    >
      1 Conversation selected
    </p>
  </section>
</div>
`;
